tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,101 @@
+    """
+    @summary: Kill a process in the specified container
+
+    """
+    running_status, process_id = get_process_id(duthost, container_name, process)
+    if running_status:
+        duthost.shell("docker exec {} kill -SIGTERM {}".format(container_name, process_id)) 
+
+    running_status, process_id = get_process_id(duthost, container_name, process)
+    if running_status:
+        assert False, "Failed to stop %s process" % process
+    else:
+        logging.info("{} process is stopped successfully".format(process)) 
+
+def check_container_status(duthost, container_name, should_be_stopped):
+    """
+    @summary: Determine whether a container should be in running state or not
+    """
+    is_running = duthost.shell("docker inspect -f \{\{.State.Running\}\} %s" % container_name)
+    if is_running['stdout'].strip() == "false" and should_be_stopped:
+        return True
+    if is_running['stdout'].strip() == "true" and not should_be_stopped:
+        return True
+    return False
+ 
+def verify_autorestart_with_critical_process(duthost, container_name, process):
+    """
+    @summary: Killing a critical process in a container to verify whether the container 
+              can be stopped and then restarted correctly 
+    """
+    
+    kill_process(duthost, container_name, process)
+
+    logging.info("Waiting until {} is stopped...".format(container_name))
+    assert wait_until(CONTAINER_STOP_TEST_WAITING_TIME, 
+                      CONTAINER_STOP_TEST_CHECK_INTERVAL,
+                      check_container_status, duthost, 
+                      container_name, True), "Failed to stop %s" % container_name
+    logging.info("{} is stopped".format(container_name))
+
+    logging.info("Waiting until {} is restarted...".format(container_name))
+    assert wait_until(CONTAINER_RESTART_TEST_WAITING_TIME, 
+                      CONTAINER_RESTART_TEST_CHECK_INTERVAL,
+                      check_container_status, duthost, 
+                      container_name, False), "Failed to restart %s" % container_name
+    logging.info("{} is restarted".format(container_name))
+
+def verify_autorestart_with_non_critical_process(duthost, container_name, process):
+    """
+    @summary: Killing a non-critical process in a container to verify whether the container 
+              is still in the running state 
+    """
+    
+    kill_process(duthost, container_name, process)
+
+    logging.info("Checking whether the {} is still running...".format(container_name))
+    assert wait_until(CONTAINER_RUNNING_TEST_WAITING_TIME, 
+                      CONTAINER_RUNNING_TEST_CHECK_INTERVAL,
+                      check_container_status, duthost, 
+                      container_name, False), "%s is stopped unexpectedly" % container_name
+    logging.info("{} is running".format(container_name))
+
+def test_swss_autorestart(duthost):
+    verify_autorestart_with_critical_process(duthost, "swss", "portsyncd")
```

Comment:
There is no check being done to ensure the auto-restart feature is enabled for the swss container before performing the test. We should ensure the feature is enabled before attempting to test it.

We should also incorporate tests later to ensure that the enabling/disabling works, also.

Reply:
Yes, I am considering to add this pre-setup to ensure this feature is enabled before doing the test. Right now I will add it after confirmation with you.